parquet-hadoop/src/test/java/org/apache/parquet/statistics/TestColumnIndexes.java

```diff
@@ -0,0 +1,306 @@
+    private static final GroupFactory FACTORY = new SimpleGroupFactory(SCHEMA);
+    private final long seed;
+    private final int pageRowCountLimit;
+    private final int columnIndexTruncateLength;
+
+    private WriteContext(long seed, int pageRowCountLimit, int columnIndexTruncateLength) {
+      this.seed = seed;
+      this.pageRowCountLimit = pageRowCountLimit;
+      this.columnIndexTruncateLength = columnIndexTruncateLength;
+    }
+
+    public Path write(Path directory) throws IOException {
+      Path file = new Path(directory, "testColumnIndexes_" + this + ".parquet");
+      Random random = new Random(seed);
+      int recordCount = random.nextInt(MAX_TOTAL_ROWS) + 1;
+      List<Supplier<?>> generators = buildGenerators(recordCount, random);
+      Configuration conf = new Configuration();
+      ParquetOutputFormat.setColumnIndexTruncateLength(conf, columnIndexTruncateLength);
+      try (ParquetWriter<Group> writer = ExampleParquetWriter.builder(file)
+          .withType(SCHEMA)
+          .withPageRowCountLimit(pageRowCountLimit)
+          .withConf(conf)
+          .build()) {
+        for (int i = 0; i < recordCount; i++) {
+          writer.write(createGroup(generators, random));
+        }
+      }
+      return file;
+    }
+
+    private Group createGroup(List<Supplier<?>> generators, Random random) {
+      Group group = FACTORY.newGroup();
+      for (int column = 0, columnCnt = SCHEMA.getFieldCount(); column < columnCnt; ++column) {
+        Type type = SCHEMA.getType(column);
+        Supplier<?> generator = generators.get(column);
+        // 2% chance of null value for an optional column
+        if (generator == null || (type.isRepetition(OPTIONAL) && random.nextInt(50) == 0)) {
+          continue;
+        }
+        switch (type.asPrimitiveType().getPrimitiveTypeName()) {
+        case BINARY:
+        case FIXED_LEN_BYTE_ARRAY:
+        case INT96:
+          group.append(type.getName(), (Binary) generator.get());
+          break;
+        case INT32:
+          group.append(type.getName(), (Integer) generator.get());
+          break;
+        case INT64:
+          group.append(type.getName(), (Long) generator.get());
+          break;
+        case FLOAT:
+          group.append(type.getName(), (Float) generator.get());
+          break;
+        case DOUBLE:
+          group.append(type.getName(), (Double) generator.get());
+          break;
+        case BOOLEAN:
+          group.append(type.getName(), (Boolean) generator.get());
+          break;
+        }
+      }
+      return group;
+    }
+
+    @Override
+    public String toString() {
+      return "seed=" + seed
+          + ",pageRowCountLimit=" + pageRowCountLimit
+          + ",columnIndexTruncateLength=" + columnIndexTruncateLength;
+    }
+  }
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TestColumnIndexes.class);
+  private static Path tmp;
+
+  @BeforeClass
+  public static void init() throws IOException {
```

Comment:
This is also part of JUnit: https://garygregory.wordpress.com/2010/01/20/junit-tip-use-rules-to-manage-temporary-files-and-folders/

This would result in less boilerplate